### ISSUE TITLE
Add a function to reset the total transfer

### ISSUE THREAD

**Issue by ItsJustMiaouss** [vladmarica/energy-meters#17]
For the moment, the Energy Meter block must be broken and replaced to reset the Total Transferred value to 0.

It would be interesting to add a `setTotalEnergyTransferred():number` or a `resetTotalEnergyTransferred()` function to avoid having to replace the block.
You can also add a button on the block interface to do this.

**Comment by amelia-bartlett** [comment]
And a way to lock it to "trusted players" or similar, so you can set up an electricity company.

**Comment by TrevorBagels** [comment]
> And a way to lock it to "trusted players" or similar, so you can set up an electricity company.

glad to hear I'm not the only one trying to charge people for their electricity usage lol

**Comment by amelia-bartlett** [comment]
> > And a way to lock it to "trusted players" or similar, so you can set up an electricity company.
> 
> glad to hear I'm not the only one trying to charge people for their electricity usage lol

lol its something i have really wanted to do for a while, i have industrial wires to make a nice control panel showing my customers

**Comment by Vaex** [comment]
This would be a great feature. A Button to Reset and deactivate the Total Transfer Rate.

**Comment by ItsJustMiaouss** [comment]
I did a fork of this repository ~2 years ago. I added these two lua functions:
- setTotalEnergyTransferred():number
- resetTotalEnergyTransferred()

You can check the repo here: https://github.com/ItsJustMiaouss/energy-meters
I hadn't made a pull request because the code wasn't very clean :)

**Comment by Vaex** [comment]
Alright thank you @ItsJustMiaouss 👍

**Comment by nikitnv** [comment]
> I did a fork of this repository ~2 years ago. I added these two lua functions:
> 
> * setTotalEnergyTransferred():number
> * resetTotalEnergyTransferred()
> 
> You can check the repo here: https://github.com/ItsJustMiaouss/energy-meters I hadn't made a pull request because the code wasn't very clean :)

Is there the way to clean it via command/command block, without lua computers mods? OC is very difficult mod for learning but I really want to have a way to reset it in my game.

**Comment by ItsJustMiaouss** [comment]
> > I did a fork of this repository ~2 years ago. I added these two lua functions:
> > 
> > * setTotalEnergyTransferred():number
> > * resetTotalEnergyTransferred()
> > 
> > You can check the repo here: https://github.com/ItsJustMiaouss/energy-meters I hadn't made a pull request because the code wasn't very clean :)
> 
> Is there the way to clean it via command/command block, without lua computers mods? OC is very difficult mod for learning but I really want to have a way to reset it in my game.

I don't think so. You can try using Computer Craft for 1.12.2 instead? Im might be easier than Open Computers (you won't need to put components into a computer etc...). But both of them are using lua to interact with the mod.

That said, I don't think lua for Minecraft is a very difficult thing. You have plenty of resources online to learn it :) (and you will be able to do many new things that are impossible with command blocks).
